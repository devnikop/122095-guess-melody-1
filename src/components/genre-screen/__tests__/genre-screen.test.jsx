import React from "react";
import renderer from "react-test-renderer";

import GenreScreen from "../genre-screen.jsx";

const mock = {
  question: {
    answers: [
      {
        genre: `rock`,
        src: `source1`,
      },
      {
        genre: `jazz`,
        src: `source2`,
      },
    ],
    genre: `rock`,
    type: `genre`,
  },
};

it(`snapshot`, () => {
  const { question } = mock;

  const tree = renderer.create(
    <GenreScreen
      question={question}
      onFormSubmit={jest.fn()}
      onInputChange={jest.fn()}
      renderAnswer={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
