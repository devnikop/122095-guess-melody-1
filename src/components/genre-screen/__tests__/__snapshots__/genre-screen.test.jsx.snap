// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Выберите 
    rock
     треки
  </h2>
  <form
    className="game__tracks"
    onSubmit={[Function]}
  >
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          defaultValue="answer-0"
          id="answer-0"
          name="answer"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="game__check"
          htmlFor="answer-0"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          defaultValue="answer-1"
          id="answer-1"
          name="answer"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="game__check"
          htmlFor="answer-1"
        >
          Отметить
        </label>
      </div>
    </div>
    <button
      className="game__submit button"
      type="submit"
    >
      Ответить
    </button>
  </form>
</section>
`;
